src/cartesian/XAxis.js

```diff
@@ -12,54 +12,188 @@ class XAxis extends Component {
   static displayName = 'XAxis';
 
   static propTypes = {
+    /**
```

Comment:
I am thinking we need to think about how to do this without touching their code? I am kinda going back and forth on this.